### DISCUSSION TITLE
Clarification and Design Inputs for Real-time BGP Monitor Project

### DISCUSSION THREAD

**Discussion by Arunjagan12** [InternetHealthReport/ihr-website#759] [Q&A] [answered]
Hi @romain-fontugne sir,

I am eager to express my interest in contributing to the Real-time BGP Monitor project for GSoC 2024. Before finalizing my proposal, I have a couple of questions that I believe will help me better align my approach with your expectations.

Dashboard Design Preferences:
Do you have any specific design preferences for the dashboard? I'm considering creating a mockup using Figma as part of my proposal, and I'd like to ensure that the design aligns with your vision for the project. Any guidance or preferences you can provide would be immensely helpful.

Customization Parameters:
What parameters should the user be able to customize in the dashboard? I would like to understand the specific aspects that you believe users would find most valuable to customize. This will aid me in proposing a well-rounded and user-centric design.

**Comment by romain-fontugne** [comment, marked as answer]
Thanks for the questions, my vision of the project is not definite, that's why we don't provide a mockup with the projects. Also discussions with candidates sometimes bring new and interesting ideas.

That said what I have in mind is basically, the user selects a prefix to monitor, then we fetch BGP data from RIS live for that prefix. The visualization could be mainly a graph that shows the AS paths, this looks like an horizontal tree that we can read from left to right. And next (or below) this main graph we can have line charts that show the number of BGP messages over time. We could have another one that shows the occurrence of ASes in the path over time, etc... basically summaries of the BGP messages seen with the RIS live api. It would be really cool if we could click on the line charts to see what the main graph looks like at a particular point in time.

**Reply by Arunjagan12** [reply]
Got it sir. I understand that the primary focus is on allowing users to select a prefix for monitoring and fetching corresponding BGP data from RIS Live. 
Sir, are there any specific performance or scalability considerations we should keep in mind when developing the visualization components, especially given the potentially large volumes of BGP data involved?
